codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -133,8 +140,15 @@ public int initialWindowSize(Http2Stream stream) {
         return state(stream).initialWindowSize();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This method must be called from the same event loop context as the {@link ChannelHandlerContext}
```

Comment:
@Scottmitch the HTTP/2 codec is composed of a lot of classes, each of which could be made to be thread-safe. If one of those classes were to be made thread-safe, the HTTP/2 codec would still not be.  It may be fine if the attempt here is to allow external access to one of those classes (e.g. the flow controller). However, if an action on the class causes an interaction with a non-threadsafe class, that would be bad.  Do you have a use-case in mind for making a thread-safe version of one of the classes?